packages/ontario-design-system-component-library/src/components/ontario-header/ontario-header.tsx

```diff
@@ -280,23 +279,67 @@ export class OntarioHeader {
 		}
 	}
 
+	@Listen('keydown', { target: 'window' })
+	handleKeyDown(event: KeyboardEvent) {
+		if (this.menuToggled && event.key === 'Escape') {
+			this.menuToggled = false;
+			this.menuButtonToggled.emit(this.menuToggled);
+			this.focusMenuButton();
+		}
+	}
+
 	/**
 	 * Logic to close the menu when anything outside the menu is clicked
 	 */
 	@Listen('click', { capture: true, target: 'window' })
 	handleClick(event: any) {
 		// if the button is clicked, return
-		if (
-			event.composedPath().includes(this.menuButton) ||
-			event.composedPath().includes(this.menuButtonDesktop) ||
-			event.composedPath().includes(this.menuButtonTablet) ||
-			event.composedPath().includes(this.menuButtonMobile)
-		) {
+		if (event.composedPath().includes(this.menuButton)) {
 			return;
 		}
 
 		// If the click was outside the current component, do the following
-		if (this.menuToggle) this.menuToggle = !this.menuToggle;
+		if (this.menuToggled) this.menuToggled = !this.menuToggled;
+	}
+
+	/**
+	 * Logic to close the menu when the focus leaves the menu
+	 */
+	@Listen('focusout', { target: 'window' })
+	handleFocusOut(event: FocusEvent) {
+		if (this.menuToggled && !this.el.contains(event.relatedTarget as Node)) {
+			this.menuToggled = false;
+			this.menuButtonToggled.emit(this.menuToggled);
+		}
+	}
+
+	/**
+	 * Logic to set breakpointDeviceState to the appropriate device when the screen resizes
+	 */
+	@Listen('resize', { target: 'window' })
+	handleResize() {
+		// screen breakpoint em values
+		const breakpoints = {
+			small: Number(OntarioBreakpointsSmall.replace('em', '')),
+			medium: Number(OntarioBreakpointsMedium.replace('em', '')),
+			large: Number(OntarioBreakpointsLarge.replace('em', '')),
+		};
+
+		const standardFontSize = 16;
+		let windowWidthPx = window.innerWidth;
+		let windowWidthEm = Math.ceil(windowWidthPx / standardFontSize);
```

Comment:
These can be `const`s.


```suggestion
		const windowWidthPx = window.innerWidth;
		const windowWidthEm = Math.ceil(windowWidthPx / standardFontSize);
```